lib/utils.js

```diff
@@ -53,3 +53,31 @@ export async function loadGetInitialProps (Component, ctx) {
   }
   return props
 }
+
+export class LockManager {
```

Comment:
`es6-promise-pool` looks you don't need to know the amount of jobs?  And you can just recreate a pool if all existing jobs done.